bioimageio/spec/shared/io.py

```diff
@@ -0,0 +1,348 @@
+            raw_node=raw_model, root_path=pathlib.Path(root_path), nodes_module=matching_cls.nodes
+        )
+        assert isinstance(model, matching_cls.nodes.Model)
+
+        return model
+
+    @classmethod
+    def export_package(
+        cls,
+        source: Union[RawModelNode, os.PathLike, str, dict],
+        root_path: Optional[os.PathLike] = None,
+        update_to_current_format: bool = False,
+        weights_formats_priorities: Optional[Sequence[current_spec.raw_nodes.WeightsFormat]] = None,
+    ) -> pathlib.Path:
+        """
+        weights_formats_priorities: If given only the first weights format present in the model is included.
+                                    If none of the prioritized weights formats is found all are included.
+        """
+        if isinstance(source, str) and source.startswith("http"):  # todo: improve remote source check
+            raise ValueError("Cannot export package from remote source")
+
+        raw_model, root_path = cls.ensure_raw_model(source, root_path, update_to_current_format)
+        io_cls = cls.get_matching_io_class(raw_model.format_version, "export")
+        package_path = io_cls._make_package_wo_conversions(
+            raw_model, root_path, weights_formats_priorities=weights_formats_priorities
+        )
+        return package_path
+
+    @classmethod
+    def _make_package_wo_conversions(
+        cls, raw_model: RawModelNode, root_path: pathlib.Path, weights_formats_priorities: Optional[Sequence[str]]
+    ):
+        package_file_name = raw_model.name
+        if raw_model.version is not missing:
+            package_file_name += f"_{raw_model.version}"
+
+        package_file_name = package_file_name.replace(" ", "_").replace(".", "_")
+
+        BIOIMAGEIO_CACHE_PATH.mkdir(exist_ok=True, parents=True)
+        package_path = (BIOIMAGEIO_CACHE_PATH / package_file_name).with_suffix(".zip")
+        max_cached_packages_with_same_name = 100
+        for p in range(max_cached_packages_with_same_name):
+            if package_path.exists():
+                package_path = (BIOIMAGEIO_CACHE_PATH / f"{package_file_name}p{p}").with_suffix(".zip")
+            else:
+                break
+        else:
+            raise FileExistsError(
+                f"Already caching {max_cached_packages_with_same_name} versions of {BIOIMAGEIO_CACHE_PATH / package_file_name}!"
+            )
+        package_content = cls._get_package_content_wo_conversions(
+            deepcopy(raw_model), root_path=root_path, weights_formats_priorities=weights_formats_priorities
+        )
+        cls.make_zip(package_path, package_content)
+        return package_path
+
+    @classmethod
+    def get_package_content(
+        cls,
+        source: Union[RawModelNode, os.PathLike, str, dict],
+        root_path: pathlib.Path,
+        update_to_current_format: bool = False,
+        weights_formats_priorities: Optional[Sequence[str]] = None,
+    ) -> Dict[str, Union[str, pathlib.Path]]:
```

Comment:
> 2) used to generate a list of files that will be uploaded to zenodo one by on

`get_package_content()` returns a dict of file names and their current local path (or their content as a string)